tests/pos/tuple-product.scala

```diff
@@ -0,0 +1,5 @@
+val a: Product = 1 *: ()
```

Comment:
This is missing testcases that actually call methods defined on Product and verify they work correctly. (Also I don't know anything about how Tuple is supposed to work, so I don't know if this change is good or bad, someone else should review it)

Reply:
I moved the test to `tests/run` and added some calls to `Product` methods. Note that in `scala.runtime.Tuple` we also have plenty of calls to `Product` on these tuples.